packages/store/addon/-private/system/snapshot.ts

```diff
@@ -25,14 +62,22 @@ function relationshipStateFor(instance, propertyName) {
   @param {Model} internalModel The model to create a snapshot from
 */
 export default class Snapshot {
-  constructor(options, identifier, store) {
-    this.__attributes = null;
-    this._belongsToRelationships = Object.create(null);
-    this._belongsToIds = Object.create(null);
-    this._hasManyRelationships = Object.create(null);
-    this._hasManyIds = Object.create(null);
-    let internalModel = (this._internalModel = store._internalModelForResource(identifier));
-    this._store = store;
+  private __attributes: Dict<any> | null = null;
```

Comment:
Could we make these `Dict<unknown>` instead? Doing so forces the check on us in at least one place: setting up `adapterOptions`, which requires you to check that it's at least not bullish, which in turn guarantees that `public adapterOptions` not being optional is actually true. 😅 

```suggestion
  private __attributes: Dict<unknown> | null = null;
```